src/Core/Form/IdentifiableObject/CommandBuilder/Accessor/CommandAccessor.php

```diff
@@ -0,0 +1,165 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\Accessor;
+
+use Symfony\Component\PropertyAccess\Exception\NoSuchIndexException;
+use Symfony\Component\PropertyAccess\Exception\NoSuchPropertyException;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+
+/**
+ * This class is inspired from the PropertyAccessor component from Symfony, but it's centered around
+ * prefilling command objects based on a config and the data available in the flattened form data array.
+ */
+class CommandAccessor
+{
+    /**
+     * @var CommandAccessorConfig
+     */
+    private $config;
+
+    /**
+     * @var PropertyAccessorInterface
+     */
+    private $propertyAccessor;
+
+    public function __construct(CommandAccessorConfig $config)
+    {
+        $this->config = $config;
+        $this->propertyAccessor = PropertyAccess::createPropertyAccessorBuilder()
```

Comment:
Isn't something we can inject?

Reply:
I don't think so, the `PropertyAccess` is a component but it doesn't exist as a service as far as I know https://symfony.com/doc/4.4/components/property_access.html

Besides, it is used internally in this class (which should also be seen as a component), but it's not really a dependency that could be adapted by different external implementation, so creating it internally makes sens to me

Reply:
I confirm, it's a util class that has IMO no advantages to be injected. Like json_encode